pkg/application/componentInstallationUseCase_test.go

```diff
@@ -2,19 +2,359 @@ package application
 
 import (
 	"context"
+	"github.com/Masterminds/semver/v3"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain"
 	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/ecosystem"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
 	"testing"
 	"time"
 )
 
+const (
+	componentName1 = "operator1"
+	testNamespace  = "k8s"
+)
+
+var (
+	semVer3212, _ = semver.NewVersion("3.2.1-2")
+)
+
 func TestNewComponentInstallationUseCase(t *testing.T) {
-	componentRepo := newMockComponentInstallationRepository(t)
-	healthConfigRepo := newMockHealthConfigProvider(t)
-	componentUseCase := NewComponentInstallationUseCase(componentRepo, healthConfigRepo)
-	assert.Same(t, componentRepo, componentUseCase.componentRepo)
-	assert.Same(t, healthConfigRepo, componentUseCase.healthConfigProvider)
+	t.Run("success", func(t *testing.T) {
+		// given
+		blueprintSpecRepoMock := newMockBlueprintSpecRepository(t)
+		componentRepoMock := newMockComponentInstallationRepository(t)
+		healthConfigRepo := newMockHealthConfigProvider(t)
+
+		// when
+		useCase := NewComponentInstallationUseCase(blueprintSpecRepoMock, componentRepoMock, healthConfigRepo)
+
+		// then
+		assert.Equal(t, blueprintSpecRepoMock, useCase.blueprintSpecRepo)
+		assert.Equal(t, componentRepoMock, useCase.componentRepo)
+		assert.Same(t, healthConfigRepo, useCase.healthConfigProvider)
+	})
+}
+
+func TestComponentInstallationUseCase_ApplyComponentStates(t *testing.T) {
+	t.Run("success with no needed action", func(t *testing.T) {
+		// given
+		blueprintSpecRepoMock := newMockBlueprintSpecRepository(t)
+		componentRepoMock := newMockComponentInstallationRepository(t)
+
+		expectedBlueprintSpec := &domain.BlueprintSpec{
+			StateDiff: domain.StateDiff{
+				ComponentDiffs: []domain.ComponentDiff{
+					{
+						Name:         componentName1,
+						NeededAction: domain.ActionNone,
+						Actual: domain.ComponentDiffState{
+							Version: semVer3212,
+						},
+						Expected: domain.ComponentDiffState{
+							Version: semVer3212,
+						},
+					},
+				},
+			},
+		}
+
+		allComponents := map[string]*ecosystem.ComponentInstallation{
+			componentName1: nil,
+		}
+
+		blueprintSpecRepoMock.EXPECT().GetById(testCtx, blueprintId).Return(expectedBlueprintSpec, nil)
+		componentRepoMock.EXPECT().GetAll(testCtx).Return(allComponents, nil)
+
+		sut := &ComponentInstallationUseCase{
+			blueprintSpecRepo: blueprintSpecRepoMock,
+			componentRepo:     componentRepoMock,
+			// TODO
+			// healthCheckInterval:
```

Comment:
What's this?

Reply:
Ups this could be an old artifact. I ll delete this.